assets/js/modules/optimize/setup.js

```diff
@@ -240,16 +240,20 @@ class OptimizeSetup extends Component {
 	renderAMPSnippet() {
 		const {
 			analyticsUseSnippet,
-			ampExperimentJSON,
 			ampExperimentJSONValidated,
 		} = this.state;
+		let { ampExperimentJSON } = this.state;
 
 		const { ampEnabled } = global.googlesitekit.admin;
 
 		if ( ! analyticsUseSnippet || ! ampEnabled ) {
 			return null;
 		}
 
+		if ( typeof ampExperimentJSON !== 'string' ) {
+			ampExperimentJSON = JSON.stringify( ampExperimentJSON );
```

Comment:
Let's be sure to stringify it in a "pretty" way for display
```suggestion
			ampExperimentJSON = JSON.stringify( ampExperimentJSON, null, 2 );
```